plugins/woocommerce/tests/php/includes/wc-stock-functions-tests.php

```diff
@@ -432,4 +432,28 @@ public function test_wc_update_product_stock_no_stock_action() {
 
 		remove_action( 'woocommerce_no_stock', $callback );
 	}
+
+	/**
+	 * @testdox The wc_trigger_stock_change_actions function should only trigger actions if the product is set
+	 *          to manage stock.
+	 */
+	public function test_wc_trigger_stock_change_actions_bails_early_for_unmanaged_stock() {
+		$action_fired = false;
+		$callback     = function () use ( &$action_fired ) {
+			$action_fired = true;
+		};
+		add_action( 'woocommerce_no_stock', $callback );
+
+		$product = WC_Helper_Product::create_simple_product();
+
```

Comment:
maybe we should also have `$product->set_manage_stock( false );`, because if the default assignment of `managed_stock` changed while product creation then this test will fail.

On second thought, that might also act as a deterrent to such a change.

Reply:
Yeah, I think it's usually fine in tests to assume defaults, as that does offer some protection against inadvertently breaking backcompat by changing a default.